pkg/ingester/flush.go

```diff
@@ -241,19 +243,19 @@ func (i *Ingester) flushUserSeries(flushQueueIndex int, userID string, fp model.
 
 	userState, ok := i.userStates.get(userID)
 	if !ok {
-		return nil
+		return "NoUser", nil
```

Comment:
This means the possible values of label `outcome` are scattered around the code.
Why not return `flushReason` and extend it with `NoUser` and `NoSeries` ?